agenda-services/src/main/java/org/exoplatform/agenda/rest/AgendaEventRest.java

```diff
@@ -1713,4 +1714,78 @@ public Response getPendingEvents(
     }
   }
 
+
+  @Path("ics")
+  @GET
+  @Produces(MediaType.TEXT_PLAIN)
+  @RolesAllowed("users")
+  @Operation(
+          summary = "Returns the ICS file generated from this event",
+          description = "Returns the ICS file generated from this event.",
+          method = "GET"
+  )
+  @ApiResponses(
+          value = {
+                  @ApiResponse(responseCode = "200", description = "Request fulfilled"),
+                  @ApiResponse(responseCode = "400", description = "Invalid query parameters"),
+                  @ApiResponse(responseCode = "500", description = "Internal server error"),
+          }
+  )
+  public Response getICSOfEvent(
+          @Parameter(description= "The event Id")
+          @QueryParam(
+                  "eventId"
+          )
+          long eventId,
+          @Parameter(description= "the identity ID of the target user")
+          @QueryParam(
+                  "identityId"
+          )
+          long identityId,
+          @Parameter(description= "the timezone ID of the target user")
+          @QueryParam(
+                  "timeZoneId"
+          )
+          String timeZoneId) {
+    try {
+      if (eventId <= 0) {
```

Comment:
How do you check the current user have access to this event ? 

As a user I must'nt be able to get the ICS of an event I'm not in.

Reply:
The function  agendaEventService.getEventById(eventId, timeZoneId, identityId) checks if the user has access to the event otherwise throws an IllegalAccessException https://github.com/exoplatform/agenda/blob/develop/agenda-api/src/main/java/org/exoplatform/agenda/service/AgendaEventService.java#L64

Reply:
yes but, in 
`Identity targetUserIdentity = identityManager.getIdentity(String.valueOf(identityId));`

identityId is provided as parameter. So I can provide the id I want, and I can provide an ID which is not mine.

Reply:
@rdenarie Yes to get an event information, you will need to provide a valid event id with the identity ID of a participant. I will remove completely the identity ID in this case, and I will get the current user from the request context, WDYT ?